packages/block-library/src/navigation-link/edit.js

```diff
@@ -372,13 +374,31 @@ function NavigationLinkEdit( {
 							'core/image',
 							'core/strikethrough',
 						] }
+						onClick={ () => {
+							if ( ! url ) {
+								setIsLinkOpen( true );
+							}
+						} }
 					/>
 					{ isLinkOpen && (
 						<Popover
 							position="bottom center"
 							onClose={ () => setIsLinkOpen( false ) }
+							className="wp-block-navigation-link__popover-link-input"
 						>
-							<LinkControl
```

Comment:
FWIW, I don't think we should stop using LinkControl, or necessarily change LinkControl to resolve #28440. The goal of that issue is really just about the interactions around a Navigation Link block when it doesn't have a link set; Showing the LinkControl when and "empty" Navigation Link is selected.